jena-arq/src/main/java/org/apache/jena/sparql/engine/http/QueryEngineHTTP.java

```diff
@@ -135,8 +135,7 @@ private QueryEngineHTTP(String serviceURI, Query query, String queryString, Http
         this.query = query;
         this.queryString = queryString;
         this.service = serviceURI;
-        // Copy the global context to freeze it.
-        this.context = new Context(ARQ.getContext());
+        this.context = ARQ.getContext();
```

Comment:
The way I would have provided this is have a `getContext` on `QueryEngineHTTP` to access  it later (or add a `Context` to the constructor but that is a bigger step at this stage).

The comment in the code is there for a reason!